### PR TITLE
feat(vector): add annotations for deployment, daemonset, statefulset

### PR DESCRIPTION
**Pull request by enicholson125**
Would it be alright to add annotations for the pod controller of Vector? 

My specific use-case is that ArgoCD deployment doesn't work with the rollWorkload option, so I would like to be able to put the stakater/reloader annotation on the deployment to roll Vector when its ConfigMap dependencies are updated.

### CODE REVIEW

File: charts/vector/values.yaml [hunk 1]
```diff
@@ -131,6 +131,9 @@ podHostNetwork: false
 # for Vector Pods.
 podSecurityContext: {}
 
+# workloadResouceAnnotations -- Set annotations on the Vector Daemonset, Deployment or Statefulset.
```
Comment (neuronull) [inline charts/vector/values.yaml:134]: Nit

```suggestion
# workloadResouceAnnotations -- Set annotations on the Vector DaemonSet, Deployment or StatefulSet.
```

**Review by neuronull — COMMENTED** [review]
Chatted with @spencergilbert about this.

The changes look good, just one nit about the proper noun capitalization.

Would you also mind editing the test

https://github.com/vectordotdev/helm-charts/blob/develop/charts/vector/ci/aggregator-all-values.yaml

, to utilize the new annotations ?

**Comment by enicholson125** [comment]
Thanks! I've added the value into the aggregators-all test and pulled in the capitalization fix.

**Comment by enicholson125** [comment]
Sorry about the test failures - hopefully all working now

**Review by neuronull — APPROVED** [review]
LGTM!

**Comment by neuronull** [comment]
Thanks for the contribution!

**Pull request merged**